x-pack/platform/plugins/shared/alerting/server/maintenance_window_client/maintenance_windows_registry.ts

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { MaintenanceWindow } from '../application/maintenance_window/types';
+
+export type CallbackType = 'post_save' | 'post_delete' | 'post_update';
+export interface CallbackEvent {
+  type: CallbackType;
+  data: MaintenanceWindow | string;
+}
+class MaintenanceWindowRegistry {
+  private callbacks: Record<string, Array<(data: CallbackEvent) => Promise<void>>> = {};
+
+  register(eventType: CallbackType, callback: (event: { type: CallbackType }) => Promise<void>) {
+    if (!this.callbacks[eventType]) {
```

Comment:
I think this can be replaced with `this.callbacks[eventType] ??= []` 

ref: [Nullish coalescing assignment](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_assignment)